### PR TITLE
Create the substitute method

### PR DESCRIPTION
**Pull request by poncovka**
Call the `substitute` method of the DNF manager to substitute variables
with their values. Remove the `_replace_vars` method of the DNF payload.

### CODE REVIEW

File: tests/nosetests/pyanaconda_tests/modules/payloads/payload/module_payload_dnf_manager_test.py [hunk 1]
```diff
@@ -553,3 +553,13 @@ def download_location_test(self):
...
         self.assertEqual(self.dnf_manager.download_location, None)
+
+    def substitute_test(self):
+        """Test the substitute method."""
+        self.assertEqual(self.dnf_manager.substitute(None), "")
+        self.assertEqual(self.dnf_manager.substitute(""), "")
+        self.assertEqual(self.dnf_manager.substitute("/"), "/")
+        self.assertEqual(self.dnf_manager.substitute("/text"), "/text")
+        self.assertEqual(self.dnf_manager.substitute("/$basearch"), "/x86_64")
+        self.assertEqual(self.dnf_manager.substitute("/$releasever"), "/rawhide")
```
Comment (jkonecny12) [inline tests/nosetests/pyanaconda_tests/modules/payloads/payload/module_payload_dnf_manager_test.py:564]: Are you sure that this test will work on branched Fedora or RHEL?

**Review by jkonecny12 — COMMENTED** [review]
Not sure about the test though.

File: tests/nosetests/pyanaconda_tests/modules/payloads/payload/module_payload_dnf_manager_test.py (hunk 1 above)
Comment (poncovka) [reply tests/nosetests/pyanaconda_tests/modules/payloads/payload/module_payload_dnf_manager_test.py]: Fixed.

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**